### PR TITLE
add babyswap on bnb

### PR DESCRIPTION
**Pull request by developerfred**
Brief comments on the purpose of your changes:

```
* add babyswap dex trades on Binance Smart Chain to dex.trades table.
* add Baby and vBaby tokens
* Updated the dex.trades seed csv file
* Modified the dex.trades model
- Updated dbt_project.yml file

```
**For Dune Engine V2**

I've checked that:

### General checks:
* [x] I tested the query on dune.com after compiling the model with dbt compile (compiled queries are written to the target directory)
* [x] I used "refs" to reference other models in this repo and "sources" to reference raw or decoded tables 
* [x] if adding a new model, I added a test
* [x] the filename is unique and ends with .sql
* [x] each sql file is a select statement and has only one view, table or function defined  
* [x] column names are `lowercase_snake_cased`
* [x] if adding a new model, I edited the dbt project YAML file with new directory path for both models and seeds (if applicable)
* [x] if wanting to expose a model in the UI (Dune data explorer), I added a post-hook in the JINJA config to add metadata (blockchains, sector/project, name and contributor Dune usernames)

### Pricing checks:
* [x] `coin_id` represents the ID of the coin on coinpaprika.com
* [x] all the coins are active on coinpaprika.com (please remove inactive ones)

### Join logic:
* [x] if joining to base table (i.e. ethereum transactions or traces), I looked to make it an inner join if possible

### Incremental logic:
* [x] I used is_incremental & not is_incremental jinja block filters on both base tables and decoded tables
  * [x] where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to base table (i.e. ethereum transactions or traces), I applied join condition where block_time >= date_trunc("day", now() - interval '1 week')
* [x] if joining to prices view, I applied join condition where minute >= date_trunc("day", now() - interval '1 week')

### CODE REVIEW

File: models/babyswap/bnb/babyswap_bnb_trades.sql [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+            CASE WHEN amount0In = 0 THEN amount1In ELSE amount0In END    AS token_sold_amount_raw,
+            cast(NULL as double)                                         AS amount_usd,
+            CASE WHEN amount0Out = 0 THEN token1 ELSE token0 END         AS token_bought_address,
+            CASE WHEN amount0In = 0 THEN token1 ELSE token0 END          AS token_sold_address,
+            t.contract_address                                           AS project_contract_address,
+            t.evt_tx_hash                                                AS tx_hash,
+            ''                                                           AS trace_address,
+            t.evt_index
+    FROM {{ source('babyswap_bnb', 'Pair_evt_Swap') }} t
+    INNER JOIN {{ source('babyswap_bnb', 'BabySwapFactory_evt_PairCreated') }} p
```
Comment (Hosuke) [inline models/babyswap/bnb/babyswap_bnb_trades.sql:31]: Hi @developerfred , please verify if these sources available on dune V2 engines as well. We may need to find the decoded tables as source first then we can develop the spell/model.

Comment (developerfred) [reply models/babyswap/bnb/babyswap_bnb_trades.sql:31]: done

**Review by Hosuke — APPROVED** [review]
To whom it may concern, I have reviewed this PR and here is the summary:

- Metadata & naming conventions look good ✅
- Config blocks match uniqueness tests in schema files ✅
- Tests and seeds added for the new models ✅
- Incremental filters added when joining large tables ✅
- Contract addresses are lowercased ✅ (I will patch a minor fix for tokens_bnb addresses)
- Inner join to large tables(bnb.transactions) as possible ✅

[Sample query results here](https://dune.com/queries/1651271?d=1)

**Comment by jeff-dude** [comment]
fixed warnings returned by the test. this should be good to go.

**Pull request merged**